### PR TITLE
[DOCS-293] Z-Wave Fingerprinting Changes

### PR DESCRIPTION
**Pull request by mckeed**
Updated the fingerprinting section of the Device Type Definition doc to explain the new Z-Wave raw description and fingerprints format.

### CODE REVIEW

File: device-type-developers-guide/definition-metadata.rst [hunk 1]
```diff
@@ -213,3 +315,12 @@ It is a best practice, however, to list the clusters in ascending order.
 
 The device can have more clusters than the fingerprint specifies, and it will still pair.
 If one of the clusters specified in the fingerprint is incorrect, the device will *not* pair.
+
+Overly-general fingerprints
++++++++++++++++++++++++++++
+
+If you wish to publish or share a device handler, you must make sure that the fingerprints do not capture other devices
```
Comment (rajkaramchedu) [inline device-type-developers-guide/definition-metadata.rst]: Minor: we typically write Device Handler in our docs.

File: device-type-developers-guide/definition-metadata.rst [hunk 2]
```diff
@@ -121,13 +121,14 @@ To define a custom command for your device handler, call the ``command()`` metho
...
 --------------
 
-When trying to connect your device to the SmartThings hub, we need a way to identify and join a particular device to the hub.
+When a ZigBee or Z-Wave device is added to the SmartThings hub, we need a way to determine which device type to assign it.
 This process is known as a "join" process, or "fingerprinting".
 
-The fingerprinting process is dependent on the type of device you are looking to pair.
-SmartThings attempts to match devices coming in based on the input and output clusters a device uses, as well as a profileId (for ZigBee) or deviceId (for Z-Wave).
-By determining what capabilities the device has, SmartThings determines what your device is.
+Device handlers define "fingerprints" to specify which devices or what kinds of devices they support.
```
Comment (rajkaramchedu) [inline device-type-developers-guide/definition-metadata.rst]: Change from Device handlers to Device Handlers

File: device-type-developers-guide/definition-metadata.rst [hunk 3]
```diff
@@ -148,62 +149,163 @@ An example of a ZigBee fingerprint definition:
...
+    implementation reasons). [1]_
+**ui:**
+    This corresponds to the Z-Wave+ User Icon type.
+
+.. [1] See `this document <http://zwavepublic.com/files/sds13740-1-z-wave-plus-device-and-command-class-types-and-defines-specificationpdf>`_ for the values of identifiers defined by the Z-Wave standard.
+
+New Z-Wave fingerprint format
++++++++++++++++++++++++++++++
+
+If you're writing a device handler for a specific device, you can base the fingerprint on the manufacturer info.
```
Comment (rajkaramchedu) [inline device-type-developers-guide/definition-metadata.rst]: Ditto

File: device-type-developers-guide/definition-metadata.rst [hunk 4]
```diff
@@ -148,62 +149,163 @@ An example of a ZigBee fingerprint definition:
...
-.. code-block:: bash
+No other parameters are required. Note that you need to add quotes and commas to the more concise raw description format
+to make it valid Groovy code.
 
-    0 0 0x2001 0 8 0x30 0x71 0x72 0x86 0x85 0x84 0x80 0x70 1 0x20
+Sometimes related products are grouped under the same 'prod' ID. In that case you can use a fingerprint without the
+'model' parameter.
 
-The fingerprint will be:
+If you are writing a general device handler that supports all devices of a certain type, you can still base the
```
Comment (rajkaramchedu) [inline device-type-developers-guide/definition-metadata.rst]: Same

File: device-type-developers-guide/definition-metadata.rst [hunk 5]
```diff
@@ -148,62 +149,163 @@ An example of a ZigBee fingerprint definition:
...
+    Z-Wave Security encapsulation.
+**ff/ui:**
+    Either of these parameters can be used to match against the corresponding fields of the raw description. It is only
+    possible to use one of the following in a single fingerprint: 'type', 'deviceId', 'ff', ui'.
+**deviceJoinName:**
+    Not used for matching. If the fingerprint matches, the device will appear to the user with this name.
+
+When multiple device fingerprints match an added Z-Wave device, they are ranked first by number of 'mfr', 'prod', and
+'model' parameters, then by the number of command classes listed, and finally by the length of the 'type', 'ff', or
+'ui' parameter. When fingerprints have the same rank, self-published device handlers take precedence over the default
```
Comment (rajkaramchedu) [inline device-type-developers-guide/definition-metadata.rst]: Same

File: device-type-developers-guide/definition-metadata.rst [hunk 6]
```diff
@@ -148,62 +149,163 @@ An example of a ZigBee fingerprint definition:
...
+
+Fingerprinting best practices
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
 Add multiple fingerprints
 +++++++++++++++++++++++++
 
-You can have multiple fingerprints.
-This is often necessary when a Device Handler should work with multiple versions of a device.
+A device handler can have multiple fingerprints in order to work with multiple versions of a device.
```
Comment (rajkaramchedu) [inline device-type-developers-guide/definition-metadata.rst]: Same

File: device-type-developers-guide/definition-metadata.rst [hunk 7]
```diff
@@ -148,62 +149,163 @@ An example of a ZigBee fingerprint definition:
...
+
+.. code-block:: groovy
+
+    fingerprint profileId: "0104", inClusters: "0000, 0003, 0004, 0005, 0006, 0008, 0702"
+    fingerprint profileId: "0104", inClusters: "0000, 0003, 0004, 0005, 0006, 0008, 0702, 0B05", outClusters: "0019", manufacturer: "sengled", model: "Z01-CIA19NAE26", deviceJoinName: "Sengled Element touch"
+    fingerprint profileId: "0104", inClusters: "0000, 0003, 0004, 0005, 0006, 0008, 0702, 0B05", outClusters: "000A, 0019", manufacturer: "Jasco Products", model: "45852", deviceJoinName: "GE Zigbee Plug-In Dimmer"
+    fingerprint profileId: "0104", inClusters: "0000, 0003, 0004, 0005, 0006, 0008, 0702, 0B05", outClusters: "000A, 0019", manufacturer: "Jasco Products", model: "45857", deviceJoinName: "GE Zigbee In-Wall Dimmer"
+
+If an added device supports the inClusters in the first fingerprint but doesn't match all the extra info in any of the
+next three, it will join with the name of the device type, in this case "ZigBee Dimmer Power."
```
Comment (rajkaramchedu) [inline device-type-developers-guide/definition-metadata.rst]: Change device type to Device Handler

File: device-type-developers-guide/definition-metadata.rst (hunk 1 above)
Comment (mckeed) [inline device-type-developers-guide/definition-metadata.rst]: I'll change it but can I ask why? It's not a proper noun and it appeared uncapitalized more often in this file specifically.

Comment (rajkaramchedu) [inline device-type-developers-guide/definition-metadata.rst]: I asked myself the same question when I was starting up. Turns out this way of writing SmartThings-specific terms is the style recommended by the style book. Thanks for making the changes :- )

Comment (rajkaramchedu) [inline device-type-developers-guide/definition-metadata.rst]: And SmartThings style book isn't exactly completely defined, so there, as far as that goes.

**Comment by jimmyjames** [comment]
Looks good, thanks for doing this!

**Pull request merged**